pkg/queue/protobuf_stats_reporter_test.go

```diff
@@ -0,0 +1,151 @@
+			reporter := NewProtobufStatsReporter(pod, test.reportingPeriod)
+			// Make the value slightly more interesting, rather than microseconds.
+			reporter.startTime = reporter.startTime.Add(-5 * time.Second)
+			reporter.Report(network.RequestStatsReport{
+				AverageConcurrency:        test.concurrency,
+				AverageProxiedConcurrency: test.proxiedConcurrency,
+				RequestCount:              test.reqCount,
+				ProxiedRequestCount:       test.proxiedReqCount,
+			})
+			stat := reporter.stat.Load().(metrics.Stat)
+			if stat.RequestCount != test.expectedReqCount {
+				t.Errorf("stat.RequestCount = %v, want %v", stat.RequestCount, test.expectedReqCount)
+			}
+			if stat.AverageConcurrentRequests != test.expectedConcurrency {
+				t.Errorf("stat.AverageConcurrentRequests = %v, want %v", stat.AverageConcurrentRequests, test.expectedConcurrency)
+			}
+			if stat.ProxiedRequestCount != test.expectedProxiedRequestCount {
+				t.Errorf("stat.ProxiedRequestCount = %v, want %v", stat.ProxiedRequestCount, test.expectedProxiedRequestCount)
+			}
+			if stat.AverageProxiedConcurrentRequests != test.expectedProxiedConcurrency {
+				t.Errorf("stat.AverageProxiedConcurrentRequests = %v, want %v", stat.AverageProxiedConcurrentRequests, test.expectedProxiedConcurrency)
+			}
+			if got := stat.ProcessUptime; got < 5.0 || got > 6.0 {
+				t.Errorf("Got %v for process uptime, wanted 5.0 <= x < 6.0", got)
+			}
+		})
+	}
+}
+
+func TestProtoHandler(t *testing.T) {
+	metricsStat := atomic.Value{}
+	metricsStat.Store(testStat)
+
+	tests := []struct {
+		name     string
+		reporter ProtobufStatsReporter
+		errorMsg string
+	}{{
+		name:     "No metrics available",
+		reporter: ProtobufStatsReporter{},
+		errorMsg: "An error has occurred while serving metrics:\n\nno metrics available yet",
+	}, {
+		name: "Metrics available",
+		reporter: ProtobufStatsReporter{
+			reportingPeriod: time.Duration(1),
+			startTime:       time.Now(),
+			stat:            metricsStat,
+			podName:         "testPod"},
+	},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			req, err := http.NewRequest(http.MethodGet, "/metrics", nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+			req.Header.Set("Accept-content", network.ProtoAcceptContent)
+			rr := httptest.NewRecorder()
+			handler := http.HandlerFunc(test.reporter.Handler().(http.HandlerFunc))
+			handler.ServeHTTP(rr, req)
+			if test.errorMsg != "" { // error case
+				expected := test.errorMsg + "\n"
+				if status := rr.Code; status != http.StatusInternalServerError {
+					t.Errorf("handler returned wrong status code: got %v want %v",
+						status, http.StatusInternalServerError)
+				}
+				if rr.Body.String() != expected {
+					t.Errorf("handler returned unexpected body: got %v want %v",
+						rr.Body.String(), expected)
+				}
+			} else { // good case, data received
+				bodyBytes, err := ioutil.ReadAll(rr.Body)
+				if err != nil {
+					t.Errorf("reading body failed: %w", err)
```

Comment:
I think the `%w` will do odd things here. It should be `%v`. `%w` only works for `fmt.Errorf`